modules/kubernetes/ingress-healthz/main.tf

```diff
@@ -31,6 +31,88 @@ resource "kubernetes_namespace" "this" {
   }
 }
 
+resource "kubernetes_network_policy" "deny_default" {
+  metadata {
+    name      = "deny-default"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress", "Egress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+
+    egress {
+      to {
+        namespace_selector {}
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+
+      ports {
+        port     = 53
+        protocol = "UDP"
+      }
+    }
+
+    egress {
+      to {
+        pod_selector {}
+      }
+    }
+  }
+}
+
+resource "kubernetes_network_policy" "allow_ingress" {
+  metadata {
+    name      = "allow-ingress"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {
+        "app.kubernetes.io/instance" = "ingress-healthz"
+        "app.kubernetes.io/name" = "nginx"
+      }
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      ports {
+        port = "8080"
```

Comment:
Instead of using port number it's not worth to use name?
I played around with the editor.cilium.io and apparently it dosen't support strings but in reality it's is.

```shell
 k explain networkpolicy.spec.ingress.ports
   port	<string>
     The port on the given protocol. This can either be a numerical or named
     port on a pod. If this field is not provided, this matches all port names
     and numbers.
```